validator-info/src/validator_info.rs

```diff
@@ -0,0 +1,520 @@
+                        )),
+                ),
+        )
+        .subcommand(
+            SubCommand::with_name("get")
+                .about("Get and parse Solana Validator info")
+                .setting(AppSettings::DisableVersion)
+                .arg(
+                    Arg::with_name("json_rpc_url")
+                        .short("u")
+                        .long("url")
+                        .value_name("URL")
+                        .takes_value(true)
+                        .default_value(JSON_RPC_URL)
+                        .validator(is_url)
+                        .help("JSON RPC URL for the solana cluster"),
+                )
+                .arg(
+                    Arg::with_name("info_keypair")
+                        .short("k")
+                        .long("info-keypair")
+                        .value_name("PATH")
+                        .takes_value(true)
+                        .required_unless_one(&["info_pubkey", "all"])
+                        .conflicts_with("all")
+                        .help("/path/to/id.json"),
+                )
+                .arg(
+                    Arg::with_name("info_pubkey")
+                        .short("p")
+                        .long("info-pubkey")
+                        .value_name("PUBKEY")
+                        .takes_value(true)
+                        .required_unless_one(&["info_keypair", "all"])
+                        .conflicts_with("all")
+                        .validator(is_pubkey)
+                        .help("The pubkey of the Validator info account"),
+                )
+                .arg(
+                    Arg::with_name("all")
+                        .short("a")
+                        .long("all")
+                        .required_unless_one(&["info_keypair", "info_pubkey"])
+                        .help("Return all current Validator info"),
+                ),
+        )
+        .get_matches();
+
+    match matches.subcommand() {
+        ("publish", Some(matches)) => {
+            let json_rpc_url = matches.value_of("json_rpc_url").unwrap();
+            let rpc_client = RpcClient::new(json_rpc_url.to_string());
+
+            // Load validator-keypair
+            let mut path = dirs::home_dir().expect("home directory");
+            let id_path = if matches.is_present("validator_keypair") {
+                matches.value_of("validator_keypair").unwrap()
+            } else {
+                path.extend(&[".config", "solana", "validator-keypair.json"]);
+                if !path.exists() {
+                    println!("No validator keypair file found. Run solana-keygen to create one.");
+                    exit(1);
+                }
+                path.to_str().unwrap()
+            };
+            let validator_keypair = read_keypair(id_path)?;
+
+            // Load validator-info keypair
+            let mut path = dirs::home_dir().expect("home directory");
```

Comment:
Couldn't we make the validator info keypair ephemeral?  It never needs to hit disk, it only needs to exist long enough to transfer update access to the validator's keypair.

Reply:
Yep, reworked